### ISSUE TITLE
[TextFields] Include label text in accessibilityLabel when textfield has content

### ISSUE THREAD

**Issue by romoore** [material-components/material-components-ios#2822]
### Overview
In the TextFields examples, when VoiceOver reads an empty text field, it reads-out the "placeholder" text. When content is entered, VoiceOver only reads the content and stops reading the "placeholder" text. When present, the accessibility values should be:

### No content, no error
* Label: placeholder, helperText

### No content, error
* Label: placeholder, error

### Content, no error
* Label: placeholder, helperText
* Value: content

### Content, error
* Label: placeholder, error
* Value: content

**Comment by romoore** [comment]
Need to confirm that this matches (or improves on) the legacy (internal) behavior.

**Comment by jverkoey** [comment]
@andrewoverton What is the current status of this issue?

**Comment by jverkoey** [comment]
> Need to confirm that this matches (or improves on) the legacy (internal) behavior.

@romoore Can you clarify what research you feel needs to be done for this issue?

**Comment by jverkoey** [comment]
@andrewoverton You mentioned there's an existing bug that is similar to this. Please link to it.

**Comment by romoore** [comment]
We need to make sure we match/improve the behavior of legacy text fields (internal implementation) when we make these changes. The research is going through all use cases on the legacy implementation and documenting the behavior to be sure we're not regressing any use cases.

**Comment by andrewoverton** [comment]
I think what's outlined in this issue description makes more sense than [what we have currently](https://github.com/material-components/material-components-ios/blob/develop/components/TextFields/src/MDCTextField.m#L783), which is a mix of various `accessibilityLabels` and `accessibilityValues` being returned as the TextField's `accessibilityValue`, with no implementation for `accessibilityLabel`.

I just want to make sure we don't do anything unexpected for clients, because that could lead to more bugs like b/110878735 being filed and the implementation changing _again_.

@meiwin, what do you think about the changes described in this issue description?

**Comment by andrewoverton** [comment]
Great! Thank you for your input @meiwin. There are a few simultaneous conversations about TextField accessibility/VoiceOver behavior happening right now and your opinion is helpful. I think we should probably be able to take the approach outlined above once a few more people get on board.

And to answer your first question, I do not think the concatenation of `accessibilityValue` and `accessibilityLabel` is happening for a reason. I think this was just written a long time ago when accessibility was more of an afterthought than it is now and most developers didn't know it as well.

**Comment by andrewoverton** [comment]
@romoore, I looked at this yesterday and found that the new suggested behavior differs a bit from what UIKit would do. With a filled text field with the placeholder text set to "This is filled and floating" and the text set to "The" the superclass gave me the following values for accessibilityLabel and accessibilityValue: 

accessibilityLabel: <nil>
accessibilityValue: "The. This is filled and floating."

With the change as we discussed it in this ticket the accessibilityLabel and accessibilityValue would be as follows:

accessibilityLabel: "This is filled and floating, Error: Uh oh! Try something else.."
accessibilityValue: "The"

The placeholder text is in the label in our version but in the value in UIKit's.

Do we really want to make this change?

**Comment by romoore** [comment]
I think the best thing at this point might be to close this issue.  We've made some changes to accessibility on MDCTextField since this issue was opened and any further changes should probably involve our internal working group and at least a small usability study.
